test_ghstack.py

```diff
@@ -2005,6 +2007,36 @@ def test_default_branch_change(self) -> None:
 rUP1 Commit 1
 rINI0 Initial commit''')
 
+    # ------------------------------------------------------------------------- #
+
+    def test_labels(self) -> None:
+        # first commit
+        self.writeFileAndAdd('file1.txt', 'A')
+        self.sh.git('commit', '-m', 'Commit 1')
+        self.sh.test_tick()
+        # ghstack
+        self.gh()
+        # second commit
+        self.writeFileAndAdd('file2.txt', 'B')
+        self.sh.git('commit', '-m', 'Commit 2')
+        self.sh.test_tick()
+        # third commit
+        self.writeFileAndAdd('file3.txt', 'C')
+        self.sh.git('commit', '-m', 'Commit 3')
+        self.sh.test_tick()
+        # ghstack with labels
+        self.gh(labels=['foo', 'bar'])
+
+        def get_labels(n: int) -> List[str]:
+            labels = self.github.get(f'repos/pytorch/pytorch/issues/{n}/labels')
+            return [label['name'] for label in labels]
+
+        # was already created before second ghstack run
+        self.assertEqual(get_labels(500), [])
```

Comment:
If there is an update to the stack, labels get applied, right? This is kind of another incarnation of the `--no-skip` bug where if you edit a commit message and use `-u` it still doesn't push unless there's also a tree change. We probably should fix it sometime. It might be better to apply labels to everything in the stack?

Reply:
Actually it's hard to tell what the behavior is when a PR in the stack is updated as it's not exercised here.

Reply:
If you want labels to only apply during PR creation, it might make more sense for the labels to be something you put in the commit message, e.g., `Labels: foo bar`. The command line flag makes it ambiguous whether or not the labels get applied to updates or not.

Reply:
@SplitInfinity and I chatted about this a bit; for this PR I explicitly documented the behavior in `--help` (it says "Add this label to any newly created pull requests"), since the idea is that (like `--draft`) it only adds them to newly created PRs, but I could change it to add the labels to every PR in the stack.

Reply:
Oh, but we messed up with `--draft`! One of ghstack's design principles is to model jf's behavior as much as possible! If I make an update and `jf submit --draft`, that update will submit as a draft. And if you do an `jf submit` without you get a non-draft submission. That being said the semantics of GitHub draft is different from Phabricator draft, but I still maintain that we got it wrong!

Reply:
Given the differences in semantics between GitHub draft and Phabricator draft, how could we have gotten it right? By naming the option differently?

Reply:
@SplitInfinity I agree that it's insoluble. I probably wouldn't have named the option differently.

I'd argue that it should be possible to toggle the draft status of a PR from `ghstack`. The only reason we don't do it is because GitHub API literally has no way to change the draft status of a PR.